packages/core/src/state/reducer.ts

```diff
@@ -20,162 +28,192 @@ export const initialState: State<unknown> = {
   lastReset: -Infinity,
 };
 
-export default function reducer(
-  state: State<unknown> | undefined,
-  action: ActionTypes,
-): State<unknown> {
-  if (!state) state = initialState;
-  switch (action.type) {
-    case GC_TYPE:
-      // inline deletes are fine as these should have 0 refcounts
-      action.entities.forEach(([key, pk]) => {
-        delete (state as any).entities[key]?.[pk];
-        delete (state as any).entityMeta[key]?.[pk];
-      });
-      action.results.forEach(fetchKey => {
-        delete (state as any).results[fetchKey];
-        delete (state as any).meta[fetchKey];
-      });
-      return state;
-    case FETCH_TYPE: {
-      const optimisticResponse = action.meta.optimisticResponse;
-      if (optimisticResponse === undefined) {
-        // If 'fetch' action reaches the reducer there are no middlewares installed to handle it
-        /* istanbul ignore next */
-        if (process.env.NODE_ENV !== 'production') {
-          console.warn(
-            'Fetch appears unhandled - you are likely missing the NetworkManager middleware',
-          );
-          console.warn(
-            'See https://resthooks.io/docs/guides/redux#indextsx for hooking up redux',
-          );
-        }
+export default function createReducer(controller: Controller) {
```

Comment:
I highly recommend hiding whitespace when viewing diff for this file. (gear icon above)